### PR TITLE
corectrl: init at 1.1.1

### LINKED ISSUE #100157: CoreCtrl
**Project description**
CoreCtrl is a Free and Open Source GNU/Linux application that allows you to control with ease your computer hardware using application profiles. It aims to be flexible, comfortable and accessible to regular users.

**Metadata**

* homepage URL: https://gitlab.com/corectrl/corectrl/
* source URL: https://gitlab.com/corectrl/corectrl/
* license: gpl3
* platforms: unix, linux

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

Closes #100157 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/corectrl/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      to control with ease your computer hardware using application profiles. It
+      aims to be flexible, comfortable and accessible to regular users.
+    '';
+    licenses = licenses.gpl3Plus;
+    platforms = platforms.linux;
+  };
+}
+# TODO: remind upstream that libdrm is not detected at configure time
+# TODO: add vulkaninfo and glxinfo to $PATH
+# TODO: http://ix.io/2AKo
```
Comment (AndersonTorres) [inline pkgs/applications/misc/corectrl/default.nix:67]: Can someone help me here, please? I have tried to test it, but the application hangs until be C-c'ed.

```kf.auth: Tried to start an invalid action:  "org.corectrl.helper.init"```

http://ix.io/2AKo

Comment (greizgh) [reply pkgs/applications/misc/corectrl/default.nix:75]: I gave a quick try and it seems that corectrl attempts to launch a helper on start (this is the dbus action you see).
Eavesdropping on the system bus (dbus-monitor --system), I've seen the following output:
`"The name org.corectrl.helper was not provided by any .service files"`.

Indeed there is no such service on my current system, but the result contains a definition in `results/corectrl/share/dbus-1/system-services/org.corectrl.helper.service`.

Not sure how to make dbus aware of this, but that's probably the missing part.

**Comment by greizgh** [comment]
@AndersonTorres I just tested this PR and it works great :+1: 

As mentioned in my previous comment, I had to add corectrl to my systemPackages in order to register the dbus config.

To be able to launch the helper without root, you'll need a running polkit-authentication-agent. That was not my case since I'm not using a full DE.

File: pkgs/applications/misc/corectrl/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/misc/corectrl/default.nix]: So the best way is to make something like a NixOS module, maybe?

Comment (AndersonTorres) [reply pkgs/applications/misc/corectrl/default.nix]: @greizgh I added vulkanInfo and glxinfo stuff

**Comment by AndersonTorres** [comment]
For now I will merge it and try to remember making a NixOS service in a future PR.

**Pull request merged**